data/src/main/java/com/linkedin/data/schema/annotation/DataSchemaRichContextTraverser.java

```diff
@@ -420,6 +421,7 @@ public String formatToErrorMessage()
     Boolean _shouldContinue = null;
```

Comment:
You should make all these variables private. Package-private is still loose a constraint.

And also create a final constructor so this cannot be extended.

Reply:
I don't think you can make a constructor final? What is the purpose?

Reply:
I mean make the class final, the goal is to avoid extending this class

Reply:
I made "TraverserContext" an interface, and made a "TraverserContextImpl" private implementation class now so it shouldn't be extended.